lib/Twig/Environment.php

```diff
@@ -314,8 +325,7 @@ public function getTemplateClass($name, $index = null)
     {
         $key = $this->getLoader()->getCacheKey($name);
         $key .= json_encode(array_keys($this->extensions));
```

Comment:
Of course, but can you describe a little bit more? `implode(array_keys(...))` and update hash on each `addExtension` call?